### PR TITLE
OYAH-Address-history

### PR DESCRIPTION
**Pull request by witoldpiecewicz**
## Description
Added
- view
- template
- form
- validation

## Todo's before PR

- [ ] Branch has been rebased against develop
- [ ] Unit tests passed (`make test`)
- [ ] PR named appropriately - see [guide](https://github.com/InformedSolutions/OFS-MORE-DevOps-Tooling/wiki/Commits-guideline)
- [ ] PR description describes the overall goals of PR commits
- [ ] Templates have been checked against the relevant user-story

## Migrations

- [ ] Null fields in models specifies default value
- [ ] You generated and applied migrations (`make migrate`)
- [ ] You updated fixtures (`make export`)

## PR Todo's

Please refer to PR [guide](https://github.com/InformedSolutions/OFS-MORE-DevOps-Tooling/wiki/Pull-requests#how-to-submit-a-pull-request)

- [ ] Specified reviewers (even if it's yourself)
- [ ] Specified assignees (those who'll merge)
- [ ] Specified labels

## PR review

Please refer to PR review [guide](https://github.com/InformedSolutions/OFS-MORE-DevOps-Tooling/wiki/Pull-requests#how-to-review-a-pull_request)

- [ ] Code syntax formatting checked
- [ ] Debug messages are absent

## PR merge

Select only one:

- [ ] Should be merged?
- [ ] Should be rebased?
- [ ] Should be squashed?

Please refer to PR merge [guide](https://github.com/InformedSolutions/OFS-MORE-DevOps-Tooling/wiki/Pull-requests#how-to-merge-a-pull_request)

### CODE REVIEW

**Comment by jenniferbrown94** [comment]
rerun jenkins build

File: application/views/applicant_address_history_summary.py [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+def address_validation(context):
+    row_errors = []
+    summary_errors = []
+    check_array = context["previous_addresses_and_gaps"]
+
+    for index in range(0, len(check_array) - 1):
+        date_difference = (
+            check_array[index + 1]["start_date"] - check_array[index]["end_date"]
+        )
+        reason = "gap" if date_difference.days > 1 else False
```
Comment (jenniferbrown94) [inline application/views/applicant_address_history_summary.py:83]: users are allowed up to a 7 day gap between addresses to account for periods where they may be staying elsewhere between moving house - can you update the logic to include this? it's all detailed on the ticket

**Review by jenniferbrown94 — CHANGES_REQUESTED** [review]

File: application/views/applicant_address_history_summary.py (hunk 1 above)
Comment (witoldpiecewicz) [reply application/views/applicant_address_history_summary.py]: fixed

**Review by jenniferbrown94 — APPROVED** [review]

**Comment by jenniferbrown94** [comment]
rerun jenkins build

**Pull request merged**